docs/plots/backends.rst

```diff
@@ -0,0 +1,384 @@
+the `~sherpa.plot.pylab_backend.PylabErrorArea` backend changes the way that
+x-errors are visualized in plots from simple error bars to a shaded area.
+
+.. literalinclude:: ../../sherpa/plot/pylab_area_backend.py
+   :language: python
+   :lines: 38-99
+
+.. todo::
+
+   Put code example in here that uses it?
+
+Testing backends and plotting code
+----------------------------------
+Currently, Sherpa does not employ pixel-level tests that compare a generated
+image pixel-by-pixel to a reference image. While testing like that guarantees
+that any and all changes are found, they are susceptible to failing for
+reasons unrelated to Sherpa, such as minor changes in the default of
+:term:`matplotlib`.
+
+Instead, the plotting tests in Sherpa fall into the following categories:
+
+  - Tests that do not depend on the output of a plotting package. Those tests
+    could check e.g. the ``prepare`` stage of a plotting object.
+    They work with any backend (since Sherpa has `sherpa.plot.backends.BasicBackend`
+    there is always a backend). For those tests, no special
+    setup is needed and they will just be run with whatever backend is loaded
+    based on the installed packages and the `.sherpa.rc` file.
+  - Tests that should work with *all* backends. In particular, the
+    `sherpa.plot.backends.IndepOnlyBackend` does not perform any plotting,
+    but it does raise a warning when plotting options that are not on the list of
+    backend-independent parameters are used. Users will never use this backend,
+    but running our tests with it ensures that we don't hardcode matplotlib-specific
+    defaults into any of our objects. In principle, most plotting tests fall into this
+    category, since most code needs to work for all backends. However, running every
+    plotting test with every possible backend can make the runtime for the tests long.
+    In practice, we thus do not run every test that we could for all backends.
+    To run a test for all backends, add the `sherpa.conftest.all_plot_backends`
+    `pytest fixture <https://docs.pytest.org/en/stable/explanation/fixtures.html>`_::
+
+      def test_this_plotting_feature(all_plot_backends):
+          ... test code here ...
+
+    This fixture will execute the test several times with different active backends.
+    Alternatively, sherpa also has a fixture that runs all backends *except* for the
+    `sherpa.plot.backends.IndepOnlyBackend`: `plot_backends`. This is useful if a test
+    makes use of options that are not on the Backend-independent list and needs to avoid the
+    extra warning that `sherpa.plot.backends.IndepOnlyBackend` would emit.
+  - Tests for a specific backend using code such as ``assert plt.gca().xlabel == "text")``.
+    These tests can either use the "require" decorators from `sherpa.utils.testing` or, if parts
+    of the test are useful for other backends as well, but skip specific statements if the wrong
+    backend is active::
+
+    >>> import numpy as np
```

Comment:
When I `make html` in `docs/` I get the following error:

```
/lagado.real/sherpa/sherpa-main/docs/plots/backends.rst:379: ERROR: Inconsistent literal block quoting.
```

but I'm not at all sure what the problem is

Reply:
It's caused by something that looks almost like, but not quite, a code block further up.